brave/brave-core
# feat(wallet): Solana Connect Status in UI

## Description 
Solana Connect Status in UI


Resolves <https://github.com/brave/brave-browser/issues/22184>

### Frontend changes

#### Wallet Panel States for Solana
1. Will show `Connected` with a `Green` dot if it is connected to a dapp.
2. Will show `Disconnected` with a `Red` dot if it is not connected to a dapp.
3. If you are on a `Solana` dapp that you have never connected to before, the `Connected / Disconnected` button will be hidden.

#### Site Permissions Panel States for Solana
1. An account will show a `Revoke` button If that account has given permission to a dapp.
2. An account will show a `Trust` button if that account has not given permission to a dapp.

### Backend changes
1. We now use `BraveWalletTabHelper` to store connected status so it can be shared with UI
3. Expose new interface `IsSolanaAccountConnected` in `WalletPanelHandler`

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Create a new wallet with 2 `Solana` accounts.
2. Go to https://raydium.io/ and click `Launch App`.
3. Do not click `Connect Wallet` yet, open the `Wallet` panel and the `Connected / Disconnected` button should be hidden.
4. Now click `Connect Wallet`, choose `Phantom` and connect your `Solana` account.
5. Open the `Wallet` panel and the `Connected` button should be visible.
6. Go back to `raydium` and click `Disconnect wallet`.
7. Open the `Wallet` panel and the `Disconnected` button should be visible.
8. Go back to `raydium` and click `Connect Wallet` again.
9. Open the `Wallet` panel and the `Connected` button should be visible.
10. Click the `Connected` button and then click `Trust` for your second account.
11. Go back to the main `Wallet` panel and switch to your `Second` solana account.
12. Refresh the page and your second wallet should be connected.
13. Go to the `Site Permissions` panel and click `Revoke` for both accounts.
14. Go back the main `Wallet` panel and the `Connected / Disconnected` button should be hidden.

https://user-images.githubusercontent.com/40611140/177605030-cb7246bc-2648-4380-94d3-ff115cfb5a1a.mov
